### PR TITLE
Add granite guardian examples using watsonx

### PR DESCRIPTION
**Pull request by martinscooper**
### Model Interaction

- [ ] **Flexible LLM platform support** The platform should be easily switchable. Use LangChain or LlamaIndex.
- [x] **Use prompt guide corresponding to the model**  For example for [Granite 3.x Language Models](https://www.ibm.com/granite/docs/models/granite/#prompt-anatomy)

### Data

- [ ] **Example data**: Follow the example data guidance.

### Notebook requirements

- [x] **Notebook outputs cleared**: Ensure all notebook outputs are cleared.
- [x] **Automated testing**: Add the recipe to the automated tests as described [here](https://github.com/ibm-granite-community/community/wiki/Automated-Validation)
- [x] **Test in Google Colab**:
    - [x] Test that it works in Google Colab (Python 3.10.12).
    - [x] Colab has its own package set and Python version, so ensure compatibility.
- [x] **Test locally**:
    - [x] Ensure the code works in a fresh Python virtual environment (venv).
- [x] **Standard access to secrets and variables** Include `!pip install git+https://github.com/ibm-granite-community/utils` in the first code cell in order to make `get_env_var` available to accessing secrets and variables in the recipe.

### Incoming References

- [x] **README.md updates**:
    - [x] Add a link to the recipe in the Table of Contents (ToC).
    - [x] Include a Colab button after that link.

### GitHub

- [x] **Commits signed**: All commits must be GPG or SSH signed.
- [x] **DCO Compliance**: Developer Certificate of Origin (DCO) applies to the code, documentation, and any example data provided. Ensure commits are signed off.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,7 +57,14 @@ The "Recipes" in the Granite Snack Cookbook showcase the essential capabilities
    <a target="_blank" href="https://colab.research.google.com/github/ibm-granite-community/granite-snack-cookbook/blob/main/recipes/Granite_Guardian/Granite_Guardian_Quick_Start.ipynb">
    <img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/>
    </a>
-
+1. [Granite Guardian 3.0 Watsonx Usage](recipes/Granite_Guardian/Granite_Guardian_Watsonx_Usage.ipynb)
+   <a target="_blank" href="https://colab.research.google.com/github/ibm-granite-community/granite-snack-cookbook/blob/main/recipes/Granite_Guardian/Granite_Guardian_Watsonx_Usage.ipynb">
+   <img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/>
+   </a>
+ 1. [Travel planner Agent](recipes/AI-Agents/travel_planner_agent.ipynb)
```
Comment (adampingel) [inline README.md:64]: This is out of order.  Please place this link in the location it is on `main`

**Review by adampingel — CHANGES_REQUESTED** [review]

**Comment by adampingel** [comment]
I'm going to request a review from @bjhargrave , who will be "lead" for the "snack" cookbook.  I do note that we are now seeing "all checks have passed" for this work.

File: README.md (hunk 1 above)
Comment (martinscooper) [reply README.md]: I accidentally duplicated it. I already removed the entry.

**Review by bjhargrave — CHANGES_REQUESTED** [review]

File: recipes/Granite_Guardian/Granite_Guardian_Watsonx_Usage.ipynb [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install git+https://github.com/ibm-granite-community/utils\n",
+    "!pip install ibm-watsonx-ai transformers"
```
[the remark this answers is not in the document]
Comment (martinscooper) [reply recipes/Granite_Guardian/Granite_Guardian_Watsonx_Usage.ipynb]: Resolved

File: recipes/Granite_Guardian/Granite_Guardian_Watsonx_Usage.ipynb [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import warnings\n",
+    "from ibm_watsonx_ai.client import APIClient\n",
+    "from ibm_watsonx_ai.foundation_models import ModelInference\n",
+    "from transformers import AutoTokenizer\n",
+    "import math\n",
+    "from ibm_granite_community.notebook_utils import get_env_var\n",
+    "import math\n",
```
[the remark this answers is not in the document]
Comment (martinscooper) [reply recipes/Granite_Guardian/Granite_Guardian_Watsonx_Usage.ipynb]: Resolved

**Review by bjhargrave — APPROVED** [review]
LGTM!

**Pull request merged**